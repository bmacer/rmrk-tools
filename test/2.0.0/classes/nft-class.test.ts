import { createCollectionMock, addChangeIssuerMock, getBobKey } from "../mocks";
import { cryptoWaitReady } from "@polkadot/util-crypto";

beforeAll(async () => {
  return await cryptoWaitReady();
});

describe("rmrk2.0.0 Collection: Create", () => {
  it("should match snapshot", async () => {
    const collection = createCollectionMock();
    expect(await collection.create()).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Collection: Create", () => {
  it("should throw error", async () => {
    const collection = createCollectionMock(1);
    try {
      await collection.create();
    } catch (e) {
      expect(e.message).toMatch(
        "An already existing collection cannot be created!"
      );
    }
  });
});

describe("rmrk2.0.0 Collection: Lock", () => {
  test.only("should match snapshot", async () => {
    const collection = createCollectionMock();
    expect(collection.lock("d43593c715a56da27d-KANARIABIRDS")).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Collection: Change issuer", () => {
  it("should match snapshot", async () => {
    const collection = createCollectionMock(1);
    expect(await collection.change_issuer(getBobKey().address)).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Collection: Create", () => {
  it("should throw error", async () => {
    const collection = createCollectionMock(0);
    try {
      await collection.change_issuer(getBobKey().address);
    } catch (e) {
      expect(e.message).toMatch(
        "This collection is new, so there's no issuer to change. If it has been deployed on chain, load the existing collection as a new instance first, then change issuer."
      );
    }
  });
});

describe("rmrk2.0.0 Collection: Add change", () => {
  it("should match snapshot", async () => {
    const collection = createCollectionMock();
    expect(await collection.addChange(addChangeIssuerMock)).toMatchSnapshot();
  });
});

describe("rmrk2.0.0 Collection: Get changes", () => {
  it("should match snapshot", async () => {
    const collection = createCollectionMock();
    await collection.addChange(addChangeIssuerMock);
    expect(await collection.getChanges()).toMatchSnapshot();
  });
});
