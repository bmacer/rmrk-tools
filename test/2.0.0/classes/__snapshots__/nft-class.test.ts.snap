// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rmrk2.0.0 Collection: Add change should match snapshot 1`] = `
Collection {
  "block": 0,
  "changes": Array [
    Object {
      "block": 0,
      "caller": "",
      "field": "",
      "new": "",
      "old": "",
      "opType": "CHANGEISSUER",
    },
  ],
  "id": "d43593c715a56da27d-KANARIABIRDS",
  "issuer": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
  "max": 0,
  "metadata": "https://some.url",
  "symbol": "KANARIABIRDS",
}
`;

exports[`rmrk2.0.0 Collection: Change issuer should match snapshot 1`] = `"RMRK::CHANGEISSUER::2.0.0::d43593c715a56da27d-KANARIABIRDS::5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"`;

exports[`rmrk2.0.0 Collection: Create should match snapshot 1`] = `"RMRK::CREATE::2.0.0::%7B%22max%22%3A0%2C%22issuer%22%3A%225GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY%22%2C%22symbol%22%3A%22KANARIABIRDS%22%2C%22id%22%3A%22d43593c715a56da27d-KANARIABIRDS%22%2C%22metadata%22%3A%22https%3A%2F%2Fsome.url%22%7D"`;

exports[`rmrk2.0.0 Collection: Lock should match snapshot 1`] = `"RMRK::LOCK::2.0.0::d43593c715a56da27d-KANARIABIRDS"`;

exports[`rmrk2.0.0 Collection: Get changes should match snapshot 1`] = `
Array [
  Object {
    "block": 0,
    "caller": "",
    "field": "",
    "new": "",
    "old": "",
    "opType": "CHANGEISSUER",
  },
]
`;
